UCSD/campus-mobile
# Implement Firebase Crashlytics on Android

## Summary

Implements Firebase Crashlytics to the Android app in order to receive detailed crash reports via Firebase through the [firebase_crashlytics](https://pub.dev/packages/firebase_crashlytics) plugin. (#676) 
~~Someone with Xcode needs to do the iOS integration, instructions [here](https://pub.dev/packages/firebase_crashlytics).~~

## Changelog

[General] [Add] - Add general code for Firebase Crashlytics
[Android] [Add] - Add Android implementation for Firebase Crashlytics

## Test Plan

In order for reports to be sent to Crashlytics while in debug mode follow these steps.

1. Insert `Crashlytics.instance.enableInDevMode = true;` into main() in main.dart
2. Force a crash somewhere using `Crashlytics.instance.crash()`
3. Run the the app not using an IDE. The widget inspector interferes with sending reports. To run without the IDE use `flutter clean` in the terminal to ensure the widget inspector is detached then use `flutter run`
4. Ensure the report is sent and received. Reports will be in Crashlytics in the Firebase console.

- [x] Report sent and received for Android
